src/auth/AuthClient.ts

```diff
@@ -0,0 +1,67 @@
+import {type Observable, lastValueFrom} from 'rxjs'
```

Comment:
Sure, it's trivial to write a migration guide for those showing how to do the same things by calling `client.request` directly.
But, should we care about adding the auth typings to `client.request`?

If we do, it would mean that:
```ts
const loginProviders = await client.request({uri: '/auth/providers'})
// loginProviders is typed to `AuthProviderResponse`
```
And the same for `client.observable`.

If we don't, then the migration guide would look like this:
```ts
import {createclient, type AuthProviderResponse} from '@sanity/client'
const client = createClient()
const loginProviders = await client.request<AuthProviderResponse>({uri: '/auth/providers'})
```
Thoughts?

Reply:
🤔 Good question. I think we should remove the typings from the client and instead export them from `@sanity/types` if they're not already there.

Reply:
I couldn't get consensus for moving typings so for now I think we should just let them stay were they are and revisit :relaxed: